pkg/istio-agent/agent.go

```diff
@@ -381,6 +383,18 @@ func (a *Agent) Run(ctx context.Context) (func(), error) {
 			if err := a.xdsProxy.initIstiodDialOptions(a); err != nil {
 				log.Warnf("Failed to init xds proxy dial options")
 			}
+
+			if a.cfg.RegenerateWorkloadCerts &&
+				// only regenerate workload certs if using istiod cert provider
+				a.secOpts.PilotCertProvider == constants.CertProviderIstiod {
+				if s := a.secretCache; s != nil {
+					log.Info("ROOTCA changed, regenerating certs")
+					s.Reset()
+					_, _ = a.getWorkloadCerts(s)
+					// Force a push to envoy
+					s.OnSecretUpdate(security.WorkloadKeyCertResourceName)
```

Comment:
duplicate since it is called in `getWorkloadCerts`

Reply:
`getWorkloadCerts` won't trigger `WorkloadKeyCertResourceName` update.

Reply:
getWorkloadCerts  ->GenerateSecret ->  `registerSecret` -> sc.OnSecretUpdate(item.ResourceName)

Reply:
https://github.com/istio/istio/blob/eb0f84cd483e23bea8dca14455602a2dca8ef8e1/security/pkg/nodeagent/cache/secretcache.go#L645-L659

there's a delayed push.

Reply:
oh